src/pages/blog/starlingx-ptp-redundant-ptp-timing-clock-sources.md

```diff
@@ -0,0 +1,92 @@
+
+# Clock selection
+
+At start, the best available clock is selected active. When the active clock degrades and doesn't meet the requirements,  another eligible clock is selected active. And when a clock recovers and meets the requirement, it becomes elible again.
+
+Depending on the clock priorities, the clock switch behavior is different. There are two broad scenarios: equal or different priorities.
+
+In a scenario where all clocks are equal priority, when the active clock deteriorate and another clock satisfies the requirements, this other clock is set active. Than, when the degraded clock recovers, it becomes eligible, but in stand by.
+
+Now, in case of different priorities. The higher priority clock has precedence over the lower one. If both are good, the higher priority one is selected active. After an outage, when the higher priority clock heal, it becomes active.
+
+The administrator can configure a stability timer to prevent multiple and consecutive clock switches, when the higher priority clock fail and come back. When the selected clock degrade, another clock takes place quickly. But, when the higher priority recovers, it will only become active after the stability time expiration.
+
+During an outage event, when all the sources fail to satisfy the requirements, one of the sources is set active as fallback. The clock with the lowest clock class is selected. As a tiebreaker, if more than one source present the same lowest local clock class, the one configured first is selected.
+
+# Applications
+
+The HA feature is totally transparent for applications using the system real-time clock source. Neither the clock selection nor the clock switching should be a concern. And no application change is required.
+
+# ptp-notification registration and reporting application
+
+The ptp-notification registration and reporting applications support multiple clock sources in compliance to O-RAN O-Cloud Notification API Specification. An aggregated clock status is also supported. The user application may need enhancements to accomodate a list of clock sources.
+
+# New alarms
+
+New alarms and events have been added to StarlingX to report the HA state.
+
+The "no source clock" alarm raises when phc2sys does not have sources that meet the specified threshold in the user configuration.
+
+The "source clock is not locked to a PRC" alarm raises when one of the HA phc2sys interfaces presents a clockClass of lesser quality than the specfied maximum clockClass in the user config. The alarm indicates the degraded interface.
+
+The "selected lower priority clock source" alarm raises when phc2sys is configured with multiple sources, and the active is not the highest priority. It indicates the highest priority clock has degraded.
+
+The "automatic source selection disabled" alarm raises when an operator locks the system real-time clock to an interface. See "status & debuging" below.
```

Comment:
```suggestion
The "automatic source selection disabled" alarm is raised when an operator locks the system real-time clock to an interface. See "status & debugging" below.
```

Reply:
This comment is still unresolved.

Reply:
This comment is still unresolved.

Reply:
It was unresolved. Corrected now in the 4th commit.